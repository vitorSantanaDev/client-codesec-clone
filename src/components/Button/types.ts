import { ReactNode, SyntheticEvent } from 'react'

export type IViewProps = {
  children: ReactNode
  handleClick?: (event?: SyntheticEvent) => void
  size?: 'large' | 'medium' | 'small'
  type?: 'button' | 'submit' | 'reset'
}
