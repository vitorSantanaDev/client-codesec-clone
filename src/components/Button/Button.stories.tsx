import { ComponentStory, ComponentMeta } from '@storybook/react'
import Button from '.'

export default {
  title: 'Button',
  component: Button,
  args: {
    children: 'Quero me cadastrar'
  }
} as ComponentMeta<typeof Button>

export const Default: ComponentStory<typeof Button> = (args) => (
  <Button {...args} />
)
