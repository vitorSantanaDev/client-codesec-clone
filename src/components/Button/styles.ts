import styled, { css } from 'styled-components'

type WrapperProps = {
  size: 'large' | 'medium' | 'small'
}

const buttonSize = {
  small: () => css`
    max-width: 190px;
  `,
  medium: () => css`
    max-width: 255px;
  `,
  large: () => css`
    max-width: 330px;
  `
}

export const Wrapper = styled.button<WrapperProps>`
  ${({ theme, size }) => css`
    width: 100%;
    ${buttonSize[size]}
    padding: calc(${theme.spacings.small} - 0.4rem);

    font-weight: 600;
    text-transform: unset;
    font-size: ${theme.font.sizes.small};

    border: none;
    border-radius: 4px;

    color: ${theme.colors.white};
    background: ${theme.colors.redGradient};
    box-shadow: -10px 15px 40px -15px ${theme.colors.redLight};
  `}
`
