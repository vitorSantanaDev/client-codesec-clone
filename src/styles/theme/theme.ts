export const theme = {
  colors: {
    redPrimary: '#E83752',
    redLight: '#FA5770',
    redDark: '#D4203B',
    redError: '#EB3D3D',
    redGradient: 'linear-gradient(243.46deg, #FA5770 16.66%, #E83752 83.34%)',
    redGradientDark:
      'linear-gradient(225.05deg, #FA5770 0%, rgba(232, 55, 82, 0) 100%)',
    redGradientLight:
      'linear-gradient(225.05deg, rgba(250, 87, 112, 0.1) 0%, rgba(232, 55, 82, 0) 50%)',
    greenSucsses: '#32D957',
    white: '#FFFFFF',
    whiteLight: '#F5F0F1',
    whiteDark: '#EDE4E5',
    black: '#121111',
    blackLight: '#1C1B1B',
    blackDark: '#0A0909',
    gray1: '#141414',
    gray2: '#2C2C2C',
    gray3: '#444444',
    gray4: '#5C5C5C',
    gray5: '#747474',
    gray6: '#8C8C8C',
    gray7: '#A3A3A3'
  },
  font: {
    family: {
      default: "'Raleway', sans-serif"
    },
    sizes: {
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
      xxhuge: '8rem'
    }
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteLarge: '(max-width: 820px)',
    ltexLarge: '(max-width: 912px)',
    lteSmall: '(max-width: 420px)',
    lteXSmall: '(max-width: 280px)'
  },
  spacings: {
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
    xxhuge: '8rem'
  }
}
