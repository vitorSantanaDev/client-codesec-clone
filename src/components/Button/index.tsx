import { ReactNode, SyntheticEvent } from 'react'
import * as S from './styles'

type ButtonProps = {
  children: ReactNode
  handleClick: (event?: SyntheticEvent) => void
  size?: 'large' | 'medium' | 'small'
}

const Button = ({ children, handleClick, size = 'medium' }: ButtonProps) => (
  <S.Wrapper size={size} onClick={handleClick}>
    {children}
  </S.Wrapper>
)

export default Button
