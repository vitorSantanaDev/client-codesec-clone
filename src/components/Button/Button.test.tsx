import { screen, fireEvent } from '@testing-library/react'
import { renderTheme } from '../../styles/theme/render-theme'
import Button from '.'

const handleClickButton = jest.fn()
describe('<Button />', () => {
  it('should render the Button component with default style', () => {
    const { container } = renderTheme(
      <Button handleClick={handleClickButton}>Entrar para turma</Button>
    )
    const button = screen.getByRole('button', { name: /entrar para turma/i })
    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'max-width': '255px'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the Button component with "small" style', () => {
    const { container } = renderTheme(
      <Button size="small" handleClick={handleClickButton}>
        Quero me cadastrar
      </Button>
    )
    const button = screen.getByRole('button', { name: /quero me cadastrar/i })
    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'max-width': '190px'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the Button component with "large" style', () => {
    const { container } = renderTheme(
      <Button size="large" handleClick={handleClickButton}>
        Quero me cadastrar
      </Button>
    )
    const button = screen.getByRole('button', { name: /quero me cadastrar/i })
    expect(button).toBeInTheDocument()
    expect(button).toHaveStyle({
      'max-width': '330px'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should call the handleClickButton function when the button is clicked', () => {
    renderTheme(<Button handleClick={handleClickButton}>Click me</Button>)
    const button = screen.getByRole('button', { name: /click me/i })
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    expect(handleClickButton).toBeCalledTimes(1)
  })
})
