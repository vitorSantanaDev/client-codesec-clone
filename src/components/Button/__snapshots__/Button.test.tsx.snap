// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the Button component with "large" style 1`] = `
.c0 {
  width: 100%;
  max-width: 330px;
  padding: calc(1.6rem - 0.4rem);
  font-weight: 600;
  text-transform: unset;
  font-size: 1.6rem;
  border: none;
  border-radius: 4px;
  color: #FFFFFF;
  background: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  box-shadow: -10px 15px 40px -15px #FA5770;
}

<button
  class="c0"
>
  Quero me cadastrar
</button>
`;

exports[`<Button /> should render the Button component with "small" style 1`] = `
.c0 {
  width: 100%;
  max-width: 190px;
  padding: calc(1.6rem - 0.4rem);
  font-weight: 600;
  text-transform: unset;
  font-size: 1.6rem;
  border: none;
  border-radius: 4px;
  color: #FFFFFF;
  background: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  box-shadow: -10px 15px 40px -15px #FA5770;
}

<button
  class="c0"
>
  Quero me cadastrar
</button>
`;

exports[`<Button /> should render the Button component with default style 1`] = `
.c0 {
  width: 100%;
  max-width: 255px;
  padding: calc(1.6rem - 0.4rem);
  font-weight: 600;
  text-transform: unset;
  font-size: 1.6rem;
  border: none;
  border-radius: 4px;
  color: #FFFFFF;
  background: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  box-shadow: -10px 15px 40px -15px #FA5770;
}

<button
  class="c0"
>
  Entrar para turma
</button>
`;
